treenodes.md

```diff
@@ -0,0 +1,182 @@
+# Varieties of Tree Nodes
```

Comment:
Can this be moved to a doc subdirectory? Or do you plan on removing this before merge?